src/Lykke.Service.OAuth/Controllers/AuthorizationController.cs

```
@@ -133,9 +149,19 @@ public IActionResult AuthorizeExternal()
             }
 
             var identity = User.Identity as ClaimsIdentity;
-
+            if (identity == null)
+            {
+                return View("Error", new OpenIdConnectMessage
+                {
+                    Error = OpenIdConnectConstants.Errors.UnauthorizedClient,
+                    ErrorDescription = "Not authorized"
+                });
+            }
             var sessionId = identity.GetClaim(OpenIdConnectConstantsExt.Claims.SessionId);
-
+            if (sessionId == null)
+            {
+                _log.Warning("Session id from ClaimsIdentity is empty.");
```

We should return an error if the session not found,

---

Will check and add error view later.